src/test/java/uk/gov/hmcts/reform/ccd/documentam/controller/advice/CaseDocumentControllerAdviceTest.java

```diff
@@ -150,4 +154,20 @@ void testHandleMethodArgumentNotValidException() throws Exception {
                 .isNotNull()
                 .satisfies(throwable -> assertThat(throwable).isInstanceOf(MethodArgumentNotValidException.class)));
     }
+
+    @Test
+    void testHandleHttpClientErrorException() {
+        final HttpClientErrorException exception = new HttpClientErrorException(HttpStatus.BAD_REQUEST,
+                                                                                "Bad Request",
+                                                                                "{\"key\":\"value\"}".getBytes(),
+                                                                                StandardCharsets.UTF_8);
+
+        final ResponseEntity<Object> responseEntity = underTest.handleHttpClientErrorException(exception);
+
+        assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+        assertEquals(HttpStatus.BAD_REQUEST.value(), responseEntity.getStatusCodeValue());
+
+        assertThatJson(responseEntity.getBody())
```

Comment:
Elsewhere in this repo we use `jsonPath` ? Could this be used here to maintain consistency and avoid pulling in a new library dependency ?

```
String expectedValue = JsonPath.parse(responseEntity.getBody().toString()).read("$.key", String.class);
assertEquals(expectedValue, "value");
```

Reply:
Can do but, personally, wouldn't be my preference.